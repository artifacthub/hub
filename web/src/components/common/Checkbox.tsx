import React from 'react';
import { useHistory } from 'react-router-dom';

interface Props {
  name: string;
  label: string;
  checked: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const CheckBox = (props: Props) => {
  const history = useHistory();
  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    history.push({
      pathname: history.location.pathname,
      search: `?tags=hola`
    });
  };

  return (
    <div className="custom-control custom-checkbox mr-sm-2 mt-2">
      <input
        type="checkbox"
        className="custom-control-input"
        id={props.name}
        onChange={handleOnChange}
        checked={props.checked}
      />
      <label className="custom-control-label" htmlFor={props.name}>{props.label}</label>
    </div>
  );
}

export default CheckBox;
