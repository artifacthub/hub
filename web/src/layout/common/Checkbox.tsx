import isUndefined from 'lodash/isUndefined';
import { ChangeEvent } from 'react';

import styles from './Checkbox.module.css';

interface Props {
  name: string;
  value: string;
  label: string | JSX.Element;
  legend?: string | number;
  checked: boolean;
  disabled?: boolean;
  className?: string;
  labelClassName?: string;
  icon?: JSX.Element;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  device: string;
}

const CheckBox = (props: Props) => {
  const handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
    if (props.onChange) {
      props.onChange(e);
    }
  };

  const id = `${props.device}-${props.name}-${props.value}`;

  return (
    <div className={`form-check me-sm-2 mb-2 ${props.className}`}>
      <input
        type="checkbox"
        className={`form-check-input ${styles.input}`}
        name={props.name}
        value={props.value}
        id={id}
        onChange={handleOnChange}
        checked={props.checked}
        aria-checked={props.checked}
        disabled={props.disabled}
        tabIndex={0}
      />
      <label
        className={`form-check-label ${styles.label} ${props.labelClassName}`}
        htmlFor={id}
        data-testid="checkboxLabel"
      >
        <div className="d-flex align-items-baseline mw-100">
          {props.icon && <span className={`me-2 position-relative ${styles.icon}`}>{props.icon}</span>}
          <span className="d-inline-block text-truncate">{props.label}</span>
          {!isUndefined(props.legend) && <small className="ps-1">({props.legend})</small>}
        </div>
      </label>
    </div>
  );
};

export default CheckBox;
