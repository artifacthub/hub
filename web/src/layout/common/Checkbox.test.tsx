import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import Checkbox from './Checkbox';

const onChangeMock = jest.fn();

const defaultProps = {
  name: 'checkbox',
  value: 'val',
  label: 'label',
  checked: false,
  onChange: onChangeMock,
};

describe('Checkbox', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Checkbox {...defaultProps} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders input and label', () => {
    const { getByTestId, getByLabelText } = render(<Checkbox {...defaultProps} />);
    const checkbox: HTMLInputElement = getByTestId('checkbox') as HTMLInputElement;
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    expect(getByLabelText(defaultProps.label)).toBeInTheDocument();
  });

  it('renders checked input with legend', () => {
    const props = {
      ...defaultProps,
      legend: 1,
      checked: true,
    };
    const { getByTestId } = render(<Checkbox {...props} />);
    const checkbox: HTMLInputElement = getByTestId('checkbox') as HTMLInputElement;
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).toBeChecked();
    const label = getByTestId('checkboxLabel');
    expect(label).toBeInTheDocument();
    expect(label).toHaveTextContent(`${props.label}(${props.legend})`);
  });

  it('calls onChange to click checkbox label', () => {
    const { getByTestId } = render(<Checkbox {...defaultProps} />);
    const label = getByTestId('checkboxLabel');
    fireEvent.click(label);
    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });
});
