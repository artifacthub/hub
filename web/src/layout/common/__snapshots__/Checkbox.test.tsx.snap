// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-check me-sm-2 mb-2 undefined"
  >
    <input
      aria-checked="false"
      class="form-check-input input"
      id="all-checkbox-val"
      name="checkbox"
      tabindex="0"
      type="checkbox"
      value="val"
    />
    <label
      class="form-check-label label undefined"
      data-testid="checkboxLabel"
      for="all-checkbox-val"
    >
      <div
        class="d-flex align-items-baseline mw-100"
      >
        <span
          class="d-inline-block text-truncate"
        >
          label
        </span>
      </div>
    </label>
  </div>
</DocumentFragment>
`;
